sharing_portal/views.py

```diff
@@ -682,7 +682,8 @@ def request_daypass(request, artifact, **kwargs):
             )
         else:
             if form.errors:
-                (messages.add_message(request, messages.ERROR, e) for e in form.errors)
+                for e in form.errors:
+                    messages.add_message(request, messages.ERROR, e)
```

Comment:
Do you know why the code doesn't work with the generator? Either way, I prefer your change, since iirc it makes the stack a log nicer in the even of an exception

Reply:
Yes! Python generator expressions are static objects, which are not executed unless iterated through. So the previous code essentially created a generator object that would then promptly be thrown out by garbage collection without getting executed. Here is a demo:

```python3
>>> def side_effect(x):
...     print(x)
...     return x + 1
... 
>>> generator = (side_effect(i) for i in range(3))
>>> # The subject of the generator expression is never executed, because the generator is just being saved as an object
>>> data = list(generator)
0
1
2
```